book/src/development/trait_checks.md

```diff
@@ -0,0 +1,112 @@
+# Trait Checking
+
+Besides [type checking](type_checks.md), we might want to examine if
+a specific type `Ty` implements certain trait when implementing a lint.
+There are three approaches to achieve this, depending on if the target trait
+that we want to examine has a [diagnostic item][diagnostic_items],
+[lang items][lang_items], or neither.
+
+## Using Diagnostic Items
+
+As explained in [rustc dev guide][rustc_dev_guide], diagnostic items
+are introduced for identifying types via [Symbols][symbol].
+
+While rustc dev guide has [a section][using_diagnostic_items] on
+how to check for a specific trait on a type `Ty`, Clippy provides
+a utils function `is_trait_method`, which simplifies the process for us.
+
+For instance, if we want to examine whether an expression implements
+the `Iterator` trait, we could simply write the following code,
+providing the `LateContext` (`cx`), our expression at hand, and
+the symbol of the trait in question:
+
+```rust
+use clippy_utils::is_trait_method;
+use rustc_hir::Expr;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_span::symbol::sym;
+
+impl LateLintPass<'_> for CheckIteratorTraitLint {
+    fn check_expr(&mut self, cx: &LateContext<'_>, expr: &Expr<'_>) {
+        if is_trait_method(cx, expr, sym::Iterator) {
+            println!("This expression implements `Iterator` trait!");
+        }
+    }
+}
+```
+
+> Note: Refer to [this index][symbol_index] for all the defined `Symbol`s.
+
+## Using Lang Items
+
+Besides diagnostic items, we can also use [`lang_items`][lang_items].
+Take a look at the documentation and we find that `LanguageItems` contains
+all defined or undefined language items both from the current crate or its
```

Comment:
> all defined or *undefined* language items

I believe this could be confusing and doesn't affect the lint implementation, IIRC. Therefore, I would leave this out :)